### ISSUE TITLE
Video and Audio are out of sync when recording a file using in_recorder_factory

### ISSUE THREAD

**Issue by felix-schroder** [whitphx/streamlit-webrtc#976]
Hello,

first of all, thank you very much for this awesome component! It is helping me a lot.

I am currently trying to record the input video and audio via in_recorder_factory. With the code below, I face an issue where the video is fine, but audio is played in very short bursts followed by sequences of silence, while in the actual recording, the audio signal should be played continuously (in sync to the video).

I have noticed that with WebRtCMode.SENDRECV, everything works fine. However, with SENDONLY, I can't find a solution to solve this issue. For my purpose, it would be highly preferable to use SENDONLY mode in order to reduce the traffic coming back from the server where the app is hosted.

Thank you!

Edit: Also the video is not recording properly after a while. E.g. after 5min of recording, I get a file with a length of 13:19:32.

```
def recorder_factory():
       return MediaRecorder("test.mp4")

streamer = webrtc_streamer(
        key="rec_streamer",
        mode=WebRtcMode.SENDONLY,
        rtc_configuration=RTC_CONFIGURATION,
        media_stream_constraints={
            "video": True,
            "audio": True},
        in_recorder_factory=recorder_factory,
    )
```

**Comment by Feel2X** [comment]
I observed that the frame dimensions and aspect ratio received from the camera are varying over time. Maybe this leads to problems during recording?
However, the frame dimensions still vary if I set constraints like below. I can't set a fixed frame size larger than 640x480.

```
media_stream_constraints={
            "video": {
                "width": {"min": 1000, "ideal": 1920, "max": 1920},
                "height": {"min": 360, "ideal": 1080},
                "frameRate": {"exact": 6}},
                "aspectRatio": {"exact": 1.777778},
            "audio": True},
```

**Comment by whitphx** [comment]
Sorry for my so late reply, and thank you for your detailed bug report.

> I have noticed that with WebRtCMode.SENDRECV, everything works fine. However, with SENDONLY, I can't find a solution to solve this issue.

With this great hint, I could find a difference between the implementation of SENDRECV and SENDONLY, and added a fix on the SENDONLY internals with #1089

Please be patient for a while until it will be released.

**Comment by whitphx** [comment]
Should be fixed with v0.44.0
